.github/workflows/continuous-integration-workflow.yml

```diff
@@ -0,0 +1,155 @@
+            cmake_build_type: Release
+            minimal: true
+          - distro: fedora_nogmx
+            toolchain: clang
+            cmake_build_type: Release
+            minimal: true
+          - distro: fedora_nogmx
+            toolchain: gnu
+            cmake_build_type: Release
+            own_gmx: true
+          - distro: latest
+            toolchain: gnu
+            cmake_build_type: Release
+            module_build: true
+          - distro: fedora_gmx2019
+            toolchain: gnu
+            cmake_build_type: Release
+          - distro: fedora_gmx2019
+            toolchain: clang
+            cmake_build_type: Release
+          - distro: fedora_gmx2019_d
+            toolchain: gnu
+            cmake_build_type: Release
+          - distro: fedora_gmx2019_d
+            toolchain: clang
+            cmake_build_type: Release
+          - distro: fedora_gmx2020
+            toolchain: gnu
+            cmake_build_type: Release
+            no_regression_testing: true
+          - distro: fedora_gmx2020
+            toolchain: clang
+            cmake_build_type: Release
+            no_regression_testing: true
+          - distro: fedora_gmx2020_d
+            toolchain: gnu
+            cmake_build_type: Release
+            no_regression_testing: true
+          - distro: fedora_gmx2020_d
+            toolchain: clang
+            cmake_build_type: Release
+            no_regression_testing: true
+          - distro: fedora_gmx9999
+            toolchain: gnu
+            cmake_build_type: Release
+            continue-on-error: true
+            no_regression_testing: true
+          - distro: fedora_gmx9999
+            toolchain: clang
+            cmake_build_type: Release
+            continue-on-error: true
+            no_regression_testing: true
+          - distro: fedora_gmx9999_d
+            toolchain: gnu
+            cmake_build_type: Release
+            continue-on-error: true
+            no_regression_testing: true
+          - distro: fedora_gmx9999_d
+            toolchain: clang
+            cmake_build_type: Release
+            continue-on-error: true
+            no_regression_testing: true
+    runs-on: ubuntu-latest
+    container: votca/buildenv:${{ matrix.distro }}
+    steps:
+      - uses: actions/checkout@master
+        with:
+          submodules: true
+      - name: Run Votca Setup
+        id: setup
+        uses: votca/actions/setup@v0
+        with:
+          distro: ${{ matrix.distro }}
+          toolchain: ${{ matrix.toolchain }}
+          minimal: ${{ matrix.minimal == true }}
+          module: ${{ matrix.module_build == true }}
+          own_gmx: ${{ matrix.own_gmx == true }}
+          regression_testing: ${{ matrix.no_regression_testing != true }}
+          coverage: ${{ matrix.coverage }}
+          cmake_build_type: ${{ matrix.cmake_build_type }}
+          ctest_args: ${{ matrix.ctest_args }}
+          cmake_args: ${{ matrix.cmake_args }}
```

Comment:
Are these all env variables? Where are they getting used? Are they being used by a script or is that what ${{ steps.setup.outputs.cmake_args }} is passing to cmake? 

I don't understand what "-uses: actions/ etc..." is doing where are the actions defined?

Reply:
You are late to the party, I just moved the actual code here https://github.com/votca/actions/tree/master/setup,
the input and output stuff is just inputs and outputs of the action as in the 
https://help.github.com/en/actions/creating-actions/creating-a-docker-container-action.